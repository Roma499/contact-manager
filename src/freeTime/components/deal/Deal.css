.deal {
  position: absolute;
  box-sizing: border-box;
  border: 2px solid rgb(22, 136, 170);
  border-radius: 4px;
  margin-left: 7px;
  width: 40%;
}
