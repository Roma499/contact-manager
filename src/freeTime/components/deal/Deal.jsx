import React, { Component } from 'react';
import { connect } from 'react-redux';
import { push } from 'connected-react-router';

import { timestampToPixels, msToTime, pixelsToTimestamp } from '../../../utils/timeConverter';
import './Deal.css';

export class Deal extends Component {
  startMouseYCoord;

  constructor(props) {
    super(props);
    this.onDragStart = this.onDragStart.bind(this);
    this.onDrag = this.onDrag.bind(this);
    this.onDoubleClick = this.props.onDoubleClick.bind(this);
  }

  render() {
    const { start, finish, pixelsInMinute } = this.props;
    const style = {
      top: timestampToPixels(start, pixelsInMinute),
      height: timestampToPixels(finish - start, pixelsInMinute)
    };
    return (
      <div
        draggable
        className="deal"
        style={style}
        onDragStart={this.onDragStart}
        onDrag={this.onDrag}
        onDragOver={this.onDragOver}
        onDoubleClick={this.onDoubleClick}
      >
        {msToTime(start)} - {msToTime(finish)}
      </div>
    );
  }
  onDragStart(e) {
    this.startMouseYCoord = e.pageY;
    e.dataTransfer.dropEffect = 'move';
  }
  onDrag(e) {
    const timeDelta = pixelsToTimestamp(this.startMouseYCoord - e.pageY, this.props.pixelsInMinute);
    const newStart = this.props.start - timeDelta;
    const newFinish = this.props.finish - timeDelta;
    const newDeal = {
      id: this.props.id,
      start: newStart,
      finish: newFinish
    };
    this.props.setDraggedDeal(newDeal);
  }
  onDragOver(e) {
    e.preventDefault();
    e.dataTransfer.dropEffect = 'none';
  }
}

function mapStateToProps(state) {
  return {
    pixelsInMinute: state.freeTimeStore.layoutConfig.pixelsInMinute
  };
}
function mapDispatchToProps(dispatch, ownProps) {
  return {
    onDoubleClick: () => dispatch(push(`/free/${ownProps.id}`))
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Deal);
