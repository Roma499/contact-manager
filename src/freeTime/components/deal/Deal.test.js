import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { Deal } from './Deal';

describe('render deal without errors', () => {
  const props = {
    pixelsInMinute: 1,
    deals: {
      id: 1,
      start: 30600000,
      finish: 31300000
    },
    setDraggedDeal: jest.fn()
  };
  it('renders without crashing', () => {
    shallow(<Deal {...props} />);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<Deal {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('contact fild FormControl prop error equal false', () => {
  //   const TextInputComponent = shallow(<Deal {...props} />).find(FormControl);
  //   expect(TextInputComponent.props().error).toEqual(false);
  // });
});
