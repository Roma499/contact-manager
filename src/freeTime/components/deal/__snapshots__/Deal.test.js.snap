// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render deal without errors renders correctly 1`] = `
<div
  className="deal"
  draggable={true}
  onDoubleClick={[Function]}
  onDrag={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  style={
    Object {
      "height": NaN,
      "top": NaN,
    }
  }
>
  NaN:NaN
   - 
  NaN:NaN
</div>
`;
